Fix datasync retry

cammellos opened the pull request:
This commit fixes a bug on the mvds library where the nextEpoch would be
incorrectly summed to the retry time, resulting in messages not being
retried, or retried much less frequently the longer the app was running.
It also updates the retry timing to backoff exponentially at multiple of
30 seconds.

https://github.com/vacp2p/mvds/pull/86

cammellos committed:
Fix datasync retry

This commit fixes a bug on the mvds library where the nextEpoch would be
incorrectly summed to the retry time, resulting in messages not being
retried, or retried much less frequently the longer the app was running.
It also updates the retry timing to backoff exponentially at multiple of
30 seconds.

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (12)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 3380e4a6 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2085/1/) | 2020-11-24 08:01:17 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201124-075844-3380e4.zip) |
| :heavy_check_mark: | 3380e4a6 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2085/1/) | 2020-11-24 08:02:35 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201124-075844-3380e4.zip) |
| :heavy_check_mark: | 3380e4a6 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2085/1/) | 2020-11-24 08:05:28 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201124-075838-3380e4.aar) |
| | | | | | | |
| :heavy_check_mark: | 00346dd4 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2085/2/) | 2020-11-24 12:39:38 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201124-123742-00346d.zip) |
| :heavy_check_mark: | 00346dd4 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2085/2/) | 2020-11-24 12:40:25 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201124-123742-00346d.zip) |
| :heavy_check_mark: | 00346dd4 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2085/2/) | 2020-11-24 12:42:35 | ~4 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201124-123737-00346d.aar) |
| | | | | | | |
| :heavy_check_mark: | c4dd0ca0 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2085/3/) | 2020-11-24 12:48:32 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201124-124621-c4dd0c.zip) |
| :heavy_check_mark: | c4dd0ca0 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2085/3/) | 2020-11-24 12:49:30 | ~3 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201124-124622-c4dd0c.zip) |
| :heavy_check_mark: | c4dd0ca0 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2085/3/) | 2020-11-24 12:53:53 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201124-124619-c4dd0c.aar) |
| | | | | | | |
| :heavy_check_mark: | 5b174b58 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2085/4/) | 2020-11-24 12:52:53 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201124-125122-5b174b.zip) |
| :heavy_check_mark: | 5b174b58 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2085/4/) | 2020-11-24 12:52:58 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201124-125122-5b174b.zip) |
| :heavy_check_mark: | 5b174b58 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2085/4/) | 2020-11-24 12:59:36 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201124-125354-5b174b.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 18165802 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2085/5/) | 2020-11-24 12:56:29 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201124-125435-181658.zip) |
| :heavy_check_mark: | 18165802 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2085/5/) | 2020-11-24 12:56:46 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201124-125432-181658.zip) |
| :heavy_check_mark: | 18165802 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2085/5/) | 2020-11-24 13:05:19 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201124-125938-181658.aar) |
| | | | | | | |
| :heavy_check_mark: | dc5342d5 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2085/6/) | 2020-11-26 15:13:00 | ~17 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201126-145520-dc5342.zip) |
| :heavy_check_mark: | dc5342d5 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2085/6/) | 2020-11-26 15:15:48 | ~20 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201126-145516-dc5342.aar) |
| :heavy_check_mark: | dc5342d5 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2085/6/) | 2020-11-26 15:24:16 | ~28 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201126-145518-dc5342.zip) |

cammellos committed:
Remove unnecessary logs & fix flaky test

cammellos committed:
Don't return an error on datasync Send

if an error is returned on the Send function, datasync will keep
  retrying a message at each epoch. If the message cannot be sent (for
  example is too large), then no messages will be sent until logout.

cammellos committed:
Bump version to 0.64.1

Samyoul commented on vendor/github.com/vacp2p/mvds/node/node.go:
```diff
@@ -545,7 +545,7 @@ func (n *Node) insertSyncState(groupID *state.GroupID, messageID state.MessageID
 func (n *Node) updateSendEpoch(s state.State) state.State {
 	s.SendCount += 1
-	s.SendEpoch += n.nextEpoch(s.SendCount, n.epoch)
+	s.SendEpoch = n.nextEpoch(s.SendCount, n.epoch)
```
Such a tiny little thing can cause all kinds of chaos. 💥

Samyoul reviewed (approved):
Looks good.

Just curious, why did you remove the debug logging?

cammellos commented:
@Samyoul I was debugging an issue and it's mostly just noise, I wanted to put them on `TRACE` but `DEBUG` is the lowest level.
